core/request.py

```diff
@@ -426,11 +426,13 @@ def processTypeHint(self, typeHint: typing.ClassVar, inValue: typing.Union[str,
             return str(f), f
         elif typeHint is bool:
             if not isinstance(inValue, str):
-                raise TypeError("Input argument to boolean typehint is not a string (probably a list)")
-            if inValue in [str(True), u"1", u"yes"]:
+                raise TypeError(f"Input argument to boolean typehint is not a str, but f{type(inValue)}")
+
+            if inValue.strip().lower() in ("true", "yes", "1"):
```

Comment:
Maybe it would be nice to support the _yes_ of the current language.

Not tested yet, but i imagine something like this:
```suggestion
            if inValue.strip().lower() in ("true", "yes", "1", translate("yes")):
```

Reply:
yes this is a fine idea, it needs a little bit of testing and more investigation.

Reply:
Okay, let's do this later. I've created the issue https://github.com/viur-framework/viur-core/issues/537